device/device.go

```diff
@@ -86,80 +86,87 @@ func newDevice(deviceID, realm, credentialsSecret string, pairingBaseURL string,
 // Connect connects the device through a goroutine
 func (d *Device) Connect(result chan<- error) {
 	go func(result chan<- error) {
+		// Let's check the channel is valid
```

Comment:
This is a radical behaviour change that I think is completely wrong. Previously, if the channel was `nil` the Connect carried forward without reporting a result. Now the operation is aborted if no channel is present.

Any reason for this? Otherwise, I think we should definitely revert that.